### DISCUSSION TITLE
existing unit binning

### DISCUSSION THREAD

**Discussion by wanyingw193** [NREL/ReEDS-2.0#116] [Q&A: Analysis]
Good afternoon, I have some questions of the existing plant binning method. 

(1) When existing units are binned, how do the online year and retirement year of the units in a bin be determined? Are they both expressed as a capacity-weighted average of the bin? 
(2) How are the retirement schedules for each unit used in ReEDS? Will the model force existing units to retire if the retirement schedule is met?
(3) What is the priority of selection when an upgrade option conflicts with an endogenous or exogenous retirement schedule? Will the model choose to upgrade prior to retirement?
(4) Is it normal for "Annual Retirements BA(GW)" in bokehpivot to be negative? If there is a negative number, which part should be checked?

Thank you!

**Comment by maxxb77** [comment]
Hi there -

(1) A portion of the bin's capacity can be retired based on the
plant/unit-specific online dates. ie the total amount of the remaining
capacity in a bin will be determined by the sum of available units'
capacity given the units' online dates

(2) Correct - the retirements for existing (init-X) and endogenously built
(new-X) vintages will retire once their retirement schedule is met. The
model allows for re-powering of certain types of technologies (see
INV_REFURB) but the only cost savings for those techs is in the avoidance
of spur line costs

(3) This is just chosen by the model - we do not assign priority to
upgraded and baseline techs when deciding on what gets retired - I can
share some model logic or discuss in a meeting how you would go about
making it so upgrades reset their lifetime. We did go back and forth on
this a bit but ultimately decided to remain consistent with the previous
version of the model.
